hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/NodesListManager.java

```diff
@@ -577,7 +594,10 @@ public boolean isUntrackedNode(String hostName) {
     Set<String> hostsList = hostDetails.getIncludedHosts();
     Set<String> excludeList = hostDetails.getExcludedHosts();
 
-    return !hostsList.isEmpty() && !hostsList.contains(hostName)
+    return (!hostsList.isEmpty() || (enableNodeUntrackedWithoutIncludePath
```

Comment:
can we refactor this and make it more readable

Reply:
directly cherry-picked from Open Source, did not want to do any changes so as to avoid merge conflicts later when we upgrade